src/tests/homework.spec.js

```diff
@@ -0,0 +1,122 @@
+import { test, expect } from '@playwright/test';
+const { v4: uuidv4 } = require('uuid');
+
+
+test('1 formular registrace', async ({ page }) => {
+  await test.step('navigace do formulare registrace', async () => {
+    await page.goto('/prihlaseni');
+    await page.getByRole('link', { name: 'Zaregistrujte se' }).click();
+    await page.locator('.col-md-10').click();
+    await page.locator('form div').filter({ hasText: 'Zaregistrovat' }).nth(1).click();
+  });
+
+  await test.step('kontrola spravneho zobrazeni', async () => {
+    await test.step('overeni zobrazeni vstupních poli', async () => {
+      await expect(page.getByLabel('Jméno a příjmení')).toBeVisible()
+      await expect(page.getByLabel('Email')).toBeVisible()
+      await expect(page.getByLabel('Heslo')).toBeVisible()
+      await expect(page.getByLabel('Kontrola hesla')).toBeVisible()
+    });
+
+    await test.step('kontrola zobrazeni oproti screenshotu', async () => {
+      const screenshot = await page.screenshot()
+      await page.setViewportSize({ width: 800, height: 600 });
+      await page.screenshot({ path: "login_page_800_600.png" });
+    });
+  });
+});
+
+function getUniqueEmail() {
+  const guid = uuidv4();
+  const email = `lenka.nova${guid}@gmail123.com`;
+  return email
+}
+
+test('2 validni registrace uzivatele ', async ({ page }) => {
+  const email = getUniqueEmail();
+  const fullname = 'Lenka Nová';
+  const password = 'Lenka123';
+
+  console.log(email);
+  console.log(fullname);
+
+  await test.step('registrace uzivatele', async () => {
+    await page.goto('/registrace');
+    await page.getByLabel('Jméno a příjmení').fill(fullname);
+    await page.getByLabel('Email').fill(email);
+    await page.getByLabel('Heslo').fill(password);
+    await page.getByLabel('Kontrola hesla').fill(password);
+    await page.getByRole('button', { name: 'Zaregistrovat' }).click();
+  });
+
+  await test.step('kontrola uspesne registrace', async () => {
+    await expect(page.locator('#navbarSupportedContent')).toContainText(fullname);
+    await page.getByText('Přihlášen').click();
+    await expect(page.getByText('Přihlášen')).toBeVisible();
+    await expect(page.locator('#navbarSupportedContent .navbar-right div.nav-item')).toContainText('Přihlášen');
+    await expect(page.locator(`#navbarSupportedContent a[title="${fullname}"]`)).toContainText(fullname);
+  });
+});
+
+test.describe('test registracniho formulare', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/registrace');
+    const fullname = 'Lenka Nová'
+    console.log(fullname);
+    await page.getByLabel('Jméno a příjmení').fill(fullname);
+
+  });
+
+  test('3 registrace uzivatele s existujicim mailem', async ({ page }) => {
+
+    const email = `existujicimail@gmail123.com`;
```

Comment:
je dobre primo pojmenovat konstantu existingEmail, at je to specificke a nikdo jiny se nemusi ptat: co je to za email?